packages/extension/src/pages/sign/components/messages/render/agoric.tsx

```diff
@@ -0,0 +1,204 @@
+import { IMessageRenderer } from "../types";
+import React, { FunctionComponent } from "react";
+import { FormattedMessage } from "react-intl";
+import { CustomIcon } from "./custom-icon";
+import { Box } from "../../../../../components/box";
+import { useStore } from "../../../../../stores";
+import { observer } from "mobx-react-lite";
+import { makeMarshal } from "../../../utils/agoric/unmarshal";
+import { LoadingIcon } from "../../../../../components/icon";
+import { displayAmount } from "../../../utils/agoric/display-amount";
+import yaml from "js-yaml";
+import { useTheme } from "styled-components";
+import { ColorPalette } from "../../../../../styles";
+
+export const AgoricWalletSpendActionMessage: IMessageRenderer = {
+  process(chainId: string, msg) {
+    if (!("type" in msg && msg.type === "swingset/WalletSpendAction")) return;
```

Comment:
Why ignore the amino form? If it is not in the amino form, it will be inconvenient for ledger users because it is not supported in the ledger cosmos app.

Reply:
I think this is the amino form, so I was ignoring non-amino by mistake, because the dapp I was testing with only uses amino. I've just added support for both now.